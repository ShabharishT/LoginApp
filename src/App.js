import React, {Component} from 'react';
import { HashRouter, Switch, Route, NavLink, Redirect } from 'react-router-dom';

import Login from './Login';
import Home from './Home';

/*
 * App Component for link navigation and
 * routing of components
 */
export default class App extends Component{
    render () {
        return (
            <div className="App">
                <HashRouter>
                    <div>
                        <div className="header">
                            <NavLink exact activeClassName="active" to="/login">Login</NavLink>
                            <NavLink activeClassName="active" to="/home">Home</NavLink>
                        </div>

                        <div className="content">
                            <Switch>
                                <Route path="/login" component={Login}/>
                                <Route path="/home" component={Home}/>
                                <Redirect exact from="/" to="/login" component={Login} />
                            </Switch>
                        </div>
                    </div>
                </HashRouter>
            </div>
        )
    }
}