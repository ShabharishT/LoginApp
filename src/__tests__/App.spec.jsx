import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

describe('App Component ', () => {

    test ('Successful render', () => {
        const component = renderer.create(<App />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});