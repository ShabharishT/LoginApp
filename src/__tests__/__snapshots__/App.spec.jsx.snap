// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component  Successful render 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="header"
    >
      <a
        aria-current="page"
        className="active"
        href="/login"
        onClick={[Function]}
        style={Object {}}
      >
        Login
      </a>
      <a
        aria-current={null}
        href="/home"
        onClick={[Function]}
      >
        Home
      </a>
    </div>
    <div
      className="content"
    >
      <form
        className="demoForm"
      >
        <h2>
          Sign In
        </h2>
        <div
          className="panel panel-default"
        >
          <div
            className="formErrors"
          >
            
            
          </div>
        </div>
        <div
          className="form-group"
        >
          <div>
            <label
              htmlFor="email"
            >
              Username
            </label>
          </div>
          <input
            className="form-control"
            name="username"
            onChange={[Function]}
            type="email"
          />
        </div>
        <br />
        <div
          className="form-group"
        >
          <div>
            <label
              htmlFor="password"
            >
              Password
            </label>
          </div>
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            type="password"
          />
        </div>
        <br />
        <button
          className="btn btn-primary"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          Login
        </button>
      </form>
    </div>
  </div>
</div>
`;
